File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
    map<int,int> mp;
    for(auto it:nums){
         mp[it]++;
    }
        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> pq;
    for (auto it : mp){
        cout<<it.first<<" "<<it.second<<endl;
        pq.push({it.second,it.first});
        if(pq.size()>k){
            pq.pop();
        }
    }
    vector<int> ans(k);
    int i=k-1;
    while(!pq.empty() && i>=0){
        ans[i]=pq.top().second;
        pq.pop();
        i--;
    }
    return ans;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> mp = new HashMap<>();
        for(int a:nums){
            mp.put(a,mp.getOrDefault(a,0)+1);
        }
        Queue<int[]> pq = new PriorityQueue<>((a,b)->a[0]-b[0]);
        for(Map.Entry<Integer,Integer> entry: mp.entrySet()){
            pq.offer(new int[]{entry.getValue(),entry.getKey()});
            while(pq.size()>k){
                pq.poll();
            }
        }
        int[] ans = new int[pq.size()];
        int i=0;
        while(!pq.isEmpty()){
            ans[i++] = pq.poll()[1];
        }
        return ans;
    }
}